### PR TITLE
feat(jq): add json convert json tfsec to rdjson

### PR DESCRIPTION
**Pull request by jplanckeel**
Hello, 

exchange in this [issue](https://github.com/reviewdog/action-tfsec/issues/16
) the links to the solutions do not come up, the idea would be to make a json converter from tfsec to rdjson with jq.

This PR install JQ and change tfsec ouput by json, parsing jq to translate to rdjson and change input format for reviewdog

## Result :

image

image

### CODE REVIEW

File: script.sh [hunk 1]
```diff
@@ -29,10 +29,16 @@ echo '::group::Preparing ...'
 
   TEMP_PATH="$(mktemp -d)"
   echo "Detected ${os} running on ${arch}, will install tools in ${TEMP_PATH}"
+  JQ_PATH="${TEMP_PATH}/jq"
   REVIEWDOG_PATH="${TEMP_PATH}/reviewdog"
   TFSEC_PATH="${TEMP_PATH}/tfsec"
 echo '::endgroup::'
 
+echo "::group:: Installing jq ... https://stedolan.github.io/jq"
```
Comment (haya14busa) [inline script.sh:37]: We do not need to incall `jq` as every Available Environment has jq by default. https://github.com/actions/virtual-environments#available-environments

**Review by haya14busa — CHANGES_REQUESTED** [review]

File: script.sh [hunk 2]
```diff
@@ -29,10 +29,16 @@ echo '::group::Preparing ...'
...
+  curl -sLo $JQ_PATH https://stedolan.github.io/jq/download/linux64/jq
+  chmod +x $JQ_PATH
+echo '::endgroup::'
```
Comment (jplanckeel) [inline script.sh:40]:
```suggestion
```

File: script.sh [hunk 3]
```diff
@@ -29,10 +29,12 @@ echo '::group::Preparing ...'
 
   TEMP_PATH="$(mktemp -d)"
   echo "Detected ${os} running on ${arch}, will install tools in ${TEMP_PATH}"
+  JQ_PATH="${TEMP_PATH}/jq"
```
Comment (jplanckeel) [inline script.sh:32]:
```suggestion
```

File: script.sh [hunk 4]
```diff
@@ -69,16 +70,17 @@ echo '::group:: Running tfsec with reviewdog 🐶 ...'
   set +Eeuo pipefail
 
   # shellcheck disable=SC2086
-  "${TFSEC_PATH}/tfsec" --format=checkstyle ${INPUT_TFSEC_FLAGS:-} . \
-    | "${REVIEWDOG_PATH}/reviewdog" -f=checkstyle \
+  "${TFSEC_PATH}/tfsec" --format=json ${INPUT_TFSEC_FLAGS:-} . \
+    | "${JQ_PATH}" -r -f "${GITHUB_ACTION_PATH}/to-rdjson.jq" \
```
Comment (jplanckeel) [inline script.sh:74]:
```suggestion
    | jq -r -f "${GITHUB_ACTION_PATH}/to-rdjson.jq" \
```

File: script.sh [hunk 5]
```diff
@@ -33,6 +33,7 @@ echo '::group::Preparing ...'
   TFSEC_PATH="${TEMP_PATH}/tfsec"
 echo '::endgroup::'
 
+
```
Comment (jplanckeel) [inline script.sh:36]:
```suggestion
```

File: script.sh (hunk 1 above)
Comment (jplanckeel) [reply script.sh]: is fixed can you re-run pipeline ? :)

Comment (haya14busa) [reply script.sh]: oh, i see. 

It's ok to include installing jq script when it's not available and it support windows/mac.

**Review by haya14busa — APPROVED** [review]
Thank for your contribution! :+1:

**Comment by review-dog** [comment]
Hi, @jplanckeel! We merged your PR to reviewdog! 🐶
Thank you for your contribution! ✨

**We just invited you to join the @reviewdog organization on GitHub.**
Accept the invite by visiting https://github.com/orgs/reviewdog/invitation.
By joining the team, you'll be a part of reviewdog community and can help the maintainance of reviewdog.

Thanks again!

**Comment by dhansen-gw** [comment]
Hi, this change is breaking the action in cases where there are no `tfsec` warnings.

### Repro steps
1. Find a terraform directory with no `tfsec` violations, confirm with the following command:
```
$ tfsec --format=json . | jq .
{
  "results": null
}
```
2. Run the command added in this PR, using the filter file in this repository:
```
$ tfsec --format=json . | jq -r -f <your_path>/to-rdjson.jq
jq: error (at <stdin>:3): Cannot iterate over null (null)
```

I am not a `jq` expert, but I believe the error is introduced here:
```
  diagnostics: .results | map({
```

Cheers!

**Pull request merged**